### PR TITLE
add scaled_linear schedule in PNDM and DDPM

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds `scaled_linear` schedule in `PNDM` and `DDPM` schedulers so those could be used with latent/stable diffusion models.

### CODE REVIEW

**Review by anton-l — APPROVED** [review]
LGTM! Also, we should probably take these outside the inits later to avoid copying schedules 😅

**Comment by damian0815** [comment]
hi, sorry to dig up an old issue. i'm hacking around on Everydream (a dreambooth-like finetuner for Stable Diffusion). i'm noticing substantial differences in outputs depending on whether the `beta_scheduler` is left at `scaled_linear` (loaded from the SD2.1 diffusers-format checkpoint) or overridden to `linear`. in particular, training with `scaled_linear` makes changes in learning rate have more wildly unpredictable effects on training losses than when it's overridden with `linear`. i'm not sure how to find out which is more "correct" - would greatly appreciate any thoughts you might have.

**Comment by patrickvonplaten** [comment]
cc @patil-suraj and @pcuenca herer maybe?

**Comment by damian0815** [comment]
ahh, nm, it doesn't seem to make that much difference, i was being confused by another hyperparameter i didn't notice was chaning. sorry for the noise!
